From a974652b462ccbd47c013d1d12bb22a55a971b29 Mon Sep 17 00:00:00 2001
From: "tellen.yu" <tellen.yu@amlogic.com>
Date: Wed, 21 May 2014 09:43:58 +0800
Subject: [PATCH 4162/5965] PD#90937

Revert "PD #86911: change for mass storage in android 4.4"

This reverts commit 681b6f0828b8eeb9d4608a515bd9d788da657624.

Conflicts:

	arch/arm/configs/meson8_defconfig
---
 drivers/usb/gadget/android.c        | 24 ++++--------------------
 drivers/usb/gadget/f_mass_storage.c | 27 +++++++++------------------
 drivers/usb/gadget/storage_common.c | 22 ----------------------
 3 files changed, 13 insertions(+), 60 deletions(-)
 mode change 100755 => 100644 drivers/usb/gadget/f_mass_storage.c
 mode change 100755 => 100644 drivers/usb/gadget/storage_common.c

diff --git a/drivers/usb/gadget/android.c b/drivers/usb/gadget/android.c
index 932130632aae..2efc241c9333 100755
--- a/drivers/usb/gadget/android.c
+++ b/drivers/usb/gadget/android.c
@@ -786,18 +786,15 @@ static int mass_storage_function_init(struct android_usb_function *f,
 {
 	struct mass_storage_function_config *config;
 	struct fsg_common *common;
-	int err, i;
+	int err;
 
 	config = kzalloc(sizeof(struct mass_storage_function_config),
 								GFP_KERNEL);
 	if (!config)
 		return -ENOMEM;
 
-	//config->fsg.nluns = 1;
-	//config->fsg.luns[0].removable = 1;
-	config->fsg.nluns = CONFIG_USB_G_ANDROID_FSG_NLUNS;
-	for (i=0; i<config->fsg.nluns; i++)
-		config->fsg.luns[i].removable = 1;
+	config->fsg.nluns = 1;
+	config->fsg.luns[0].removable = 1;
 
 	common = fsg_common_init(NULL, cdev, &config->fsg);
 	if (IS_ERR(common)) {
@@ -805,26 +802,13 @@ static int mass_storage_function_init(struct android_usb_function *f,
 		return PTR_ERR(common);
 	}
 
-	/*
 	err = sysfs_create_link(&f->dev->kobj,
 				&common->luns[0].dev.kobj,
 				"lun");
 	if (err) {
 		kfree(config);
 		return err;
-	}*/
-
-	for (i=0; i<config->fsg.nluns; i++) {
-		char linkname[5];
-		snprintf(linkname, sizeof(linkname), "lun%d", i);
-		err = sysfs_create_link(&f->dev->kobj,
-					&common->luns[i].dev.kobj,
-					linkname);
-		if (err) {
-			kfree(config);
-			return err;
-		}
- 	}
+	}
 
 	config->common = common;
 	f->config = config;
diff --git a/drivers/usb/gadget/f_mass_storage.c b/drivers/usb/gadget/f_mass_storage.c
old mode 100755
new mode 100644
index 2fa246e3b45a..c35a9ecc576b
--- a/drivers/usb/gadget/f_mass_storage.c
+++ b/drivers/usb/gadget/f_mass_storage.c
@@ -789,8 +789,6 @@ static int do_write(struct fsg_common *common)
 	}
 	spin_lock(&curlun->filp->f_lock);
 	curlun->filp->f_flags &= ~O_SYNC;	/* Default is not to wait */
-	if (curlun->force_sync)
-		curlun->filp->f_flags |= O_SYNC;
 	spin_unlock(&curlun->filp->f_lock);
 
 	/*
@@ -2583,7 +2581,7 @@ static int fsg_main_thread(void *common_)
 static DEVICE_ATTR(ro, 0644, fsg_show_ro, fsg_store_ro);
 static DEVICE_ATTR(nofua, 0644, fsg_show_nofua, fsg_store_nofua);
 static DEVICE_ATTR(file, 0644, fsg_show_file, fsg_store_file);
-static DEVICE_ATTR(force_sync, 0644, fsg_show_force_sync, fsg_store_force_sync);
+
 static struct device_attribute dev_attr_ro_cdrom =
 	__ATTR(ro, 0444, fsg_show_ro, NULL);
 static struct device_attribute dev_attr_file_nonremovable =
@@ -2712,9 +2710,6 @@ static struct fsg_common *fsg_common_init(struct fsg_common *common,
 		if (rc)
 			goto error_luns;
 		rc = device_create_file(&curlun->dev, &dev_attr_nofua);
-		if (rc)
-			goto error_luns;
-		rc = device_create_file(&curlun->dev, &dev_attr_force_sync);
 		if (rc)
 			goto error_luns;
 
@@ -2748,14 +2743,13 @@ buffhds_first_it:
 
 	/* Prepare inquiryString */
 	i = get_default_bcdDevice();
-	snprintf(common->inquiry_string, sizeof common->inquiry_string,"%s","ONDA");
-	//snprintf(common->inquiry_string, sizeof common->inquiry_string,
-	//	 "%-8s%-16s%04x", cfg->vendor_name ?: "Linux",
+	snprintf(common->inquiry_string, sizeof common->inquiry_string,
+		 "%-8s%-16s%04x", cfg->vendor_name ?: "Linux",
 		 /* Assume product name dependent on the first LUN */
-	//	 cfg->product_name ?: (common->luns->cdrom
-	//			     ? "File-Stor Gadget"
-	//			     : "File-CD Gadget"),
-	//	 i);
+		 cfg->product_name ?: (common->luns->cdrom
+				     ? "File-Stor Gadget"
+				     : "File-CD Gadget"),
+		 i);
 
 	/*
 	 * Some peripheral controllers are known not to be able to
@@ -2844,7 +2838,6 @@ static void fsg_common_release(struct kref *ref)
 					   lun->removable
 					 ? &dev_attr_file
 					 : &dev_attr_file_nonremovable);
-			device_remove_file(&lun->dev, &dev_attr_force_sync);
 			fsg_lun_close(lun);
 			device_unregister(&lun->dev);
 		}
@@ -2998,7 +2991,6 @@ struct fsg_module_parameters {
 	bool		removable[FSG_MAX_LUNS];
 	bool		cdrom[FSG_MAX_LUNS];
 	bool		nofua[FSG_MAX_LUNS];
-	int		force_sync[FSG_MAX_LUNS];
 
 	unsigned int	file_count, ro_count, removable_count, cdrom_count;
 	unsigned int	nofua_count;
@@ -3031,9 +3023,8 @@ struct fsg_module_parameters {
 	_FSG_MODULE_PARAM(prefix, params, luns, uint,			\
 			  "number of LUNs");				\
 	_FSG_MODULE_PARAM(prefix, params, stall, bool,			\
-			  "false to prevent bulk stalls")				\
-	_FSG_MODULE_PARAM_ARRAY(prefix, params, force_sync, bool,	\
-				"true to force sync when do write");
+			  "false to prevent bulk stalls")
+
 static void
 fsg_config_from_params(struct fsg_config *cfg,
 		       const struct fsg_module_parameters *params)
diff --git a/drivers/usb/gadget/storage_common.c b/drivers/usb/gadget/storage_common.c
old mode 100755
new mode 100644
index 0908f7a1f8b9..dbce3a9074e6
--- a/drivers/usb/gadget/storage_common.c
+++ b/drivers/usb/gadget/storage_common.c
@@ -133,7 +133,6 @@ struct fsg_lun {
 	unsigned int	registered:1;
 	unsigned int	info_valid:1;
 	unsigned int	nofua:1;
-	unsigned int	force_sync:1;
 
 	u32		sense_data;
 	u32		sense_data_info;
@@ -593,13 +592,6 @@ static ssize_t fsg_show_file(struct device *dev, struct device_attribute *attr,
 	return rc;
 }
 
-static ssize_t fsg_show_force_sync(struct device *dev, struct device_attribute *attr,
-			      char *buf)
-{
-	struct fsg_lun	*curlun = fsg_lun_from_dev(dev);
-
-	return sprintf(buf, "%u\n", curlun->force_sync);
-}
 
 static ssize_t fsg_store_ro(struct device *dev, struct device_attribute *attr,
 			    const char *buf, size_t count)
@@ -683,17 +675,3 @@ static ssize_t fsg_store_file(struct device *dev, struct device_attribute *attr,
 	up_write(filesem);
 	return (rc < 0 ? rc : count);
 }
-static ssize_t fsg_store_force_sync(struct device *dev,
-			       struct device_attribute *attr,
-			       const char *buf, size_t count)
-{
-	struct fsg_lun	*curlun = fsg_lun_from_dev(dev);
-	unsigned long	force_sync;
-
-	if (strict_strtoul(buf, 2, &force_sync))
-		return -EINVAL;
-
-	curlun->force_sync = force_sync;
-
-	return count;
-}
-- 
2.19.0

